src/app/state/persist/syncTabs.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  AnyAction,
+  configureStore,
+  ConfigureStoreOptions,
+  Dispatch,
+  EnhancedStore,
+  Middleware,
+} from '@reduxjs/toolkit'
+import { networkActions } from 'app/state/network'
+import { isSyncingTabsSupported, needsSyncingTabs, persistActions } from 'app/state/persist'
+import { walletActions } from 'app/state/wallet'
+import { themeActions } from 'styles/theme/slice'
+import {
+  createStateSyncMiddleware,
+  initStateWithPrevTab,
+  Config as StateSyncConfig,
+  withReduxStateSync,
+} from 'redux-state-sync'
+import { RootState } from 'types'
+import { SyncedRootState } from 'app/state/persist/types'
+
+/**
+ * When opening a second tab it initially syncs these state slices.
+ */
+export function receiveInitialTabSyncState(
+  prevState: RootState,
+  initialSyncState: SyncedRootState,
+): RootState {
+  return {
+    ...prevState,
+    theme: initialSyncState.theme,
+    wallet: initialSyncState.wallet,
+    network: initialSyncState.network,
+    persist: initialSyncState.persist,
+  }
+}
+
+/**
+ * When interacting with a second tab it syncs these actions.
+ */
+export const whitelistTabSyncActions = [
+  themeActions.changeTheme.type,
+  walletActions.walletOpened.type,
+  walletActions.updateBalance.type,
+  networkActions.networkSelected.type,
```

Comment:
> @csillag: I see that it's possible to use two different accounts in two different tabs, but I can't use Mainnet on one tab and Testnet on the other. Is that an intentional choice?

Yeah. I'm syncing `networkSelected` here. It would take quite some refactoring to keep it separate. I'm syncing wallet slice and it includes balances, so after switching networks it showed testnet balance on mainnet tab.

Related note in https://github.com/oasisprotocol/oasis-wallet-web/issues/792#issuecomment-1238715243
> selected network needs to be synced as long as network-specific balance is in state.wallets[*].balance